iptables/iptables.go

```diff
@@ -241,6 +241,24 @@ func (ipt *IPTables) DeleteIfExists(table, chain string, rulespec ...string) err
 	return err
 }
 
+// DeleteById deletes the rule with the specified ID in the given table and chain.
+func (ipt *IPTables) DeleteById(table, chain string, id int) (string, error) {
+	rules, err := ipt.List(table, chain)
+	if err != nil {
+		return "", err
+	}
+
+	for _, rule := range rules {
```

Comment:
Looping all the rules here does not make any sense when target is remove just one rule with ID. Opened #125 which contains corrected version of this together with tests.

Reply:
Thanks for working on this and adding tests! I will go ahead and close this now in favor of #125.